lib/rules/named-grid-areas-no-invalid/index.js

```diff
@@ -0,0 +1,129 @@
+// @ts-nocheck
+
+'use strict';
+
+const _ = require('lodash');
+const declarationValueIndex = require('../../utils/declarationValueIndex');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const valueParser = require('postcss-value-parser');
+
+const ruleName = 'named-grid-areas-no-invalid';
+
+const messages = ruleMessages(ruleName, {
+	expectedToken: () => 'Expected cell token within string',
+	expectedSameNumber: () => 'Expected same number of cell tokens in each string',
+	expectedRectangle: (name) => `Expected single filled-in rectangle for "${name}"`,
+});
+
+function rule(actual) {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual });
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkDecls('grid-template-areas', (decl) => {
+			function complain(message, sourceIndex = 0) {
+				report({
+					message,
+					node: decl,
+					index: declarationValueIndex(decl) + sourceIndex,
+					result,
+					ruleName,
+				});
+			}
+
+			const value = decl.value;
+
+			if (value.toLowerCase().trim() === 'none') return;
+
+			const parsedValue = valueParser(value);
+
+			let areas = [];
+			let reportSent = false;
+
+			parsedValue.walk((node) => {
+				if (node.type === 'string') {
```

Comment:
Let's return early when walking (unless reporting):

```js
if (type !== 'string') return;
if (value === '') {
 complain(..)
}
areas.push(..)
```

This will be consistent with other rules.